qiskit/_compiler.py

```diff
@@ -279,6 +293,57 @@ def load_unroll_qasm_file(filename, basis_gates='u1,u2,u3,cx,id'):
     return circuit_unrolled
 
 
+def best_subset(backend, n_qubits):
+    """Computes the qubit mapping with the best
+    connectivity.
+
+    Parameters:
+        backend (Qiskit.BaseBackend): A QISKit backend instance.
+        n_qubits (int): Number of subset qubits to consider.
+
+    Returns:
+        ndarray: Array of qubits to use for best
+                connectivity mapping.
+
+    Raises:
+        ValueError: Wrong number of qubits given.
+    """
+    device_qubits = backend.configuration['n_qubits']
+    cmap = np.asarray(backend.configuration['coupling_map'])
```

Comment:
Can we move this line before all the checks? so if something is failing in the preconditions there's no need to compute this.

Reply:
Done.